src/identity/actions/thunks/index.ts

```diff
@@ -0,0 +1,121 @@
+import {GetState} from 'src/types'
+import {Dispatch} from 'react'
+
+// Actions
+import {
+  setQuartzMe,
+  setQuartzMeStatus,
+  Actions as MeActions,
+} from 'src/me/actions/creators'
+import {
+  setCurrentOrgDetails,
+  setCurrentBillingProvider,
+  setQuartzIdentity,
+  setQuartzIdentityStatus,
+  Actions as IdentityActions,
+} from 'src/identity/actions/creators'
+
+// Types
+import {RemoteDataState} from 'src/types'
+enum RetrievalError {
+  accountIdentity = 'Unable to retrieve account ID from user identity.',
+  orgIdentity = 'Unable to retrieve organization ID from user identity.',
+}
+
+// Utilities
+import {
+  convertIdentityToMe,
+  fetchQuartzIdentity,
+  fetchAccountDetails,
+  fetchOrgDetails,
+} from 'src/identity/apis/auth'
+
+// Retrieves user's quartz identity from /quartz/identity, and stores it in state.identity.
+export const getQuartzIdentityThunk = () => async (
+  dispatch: Dispatch<IdentityActions | MeActions>
+) => {
+  try {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Loading))
+
+    const quartzIdentity = await fetchQuartzIdentity()
+
+    dispatch(setQuartzIdentity(quartzIdentity))
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Done))
+
+    const legacyMe = convertIdentityToMe(quartzIdentity)
+    // Remove lines below once quartzMe is deprecated.
```

Comment:
I'm not a big fan of comments like this because they tend to overstate their authority. From what we know right now, we'll want to remove those lines when quartzMe is deprecated. But maybe something changes between now and then and this comment becomes outdated and is incorrect. If that happens, there is a vanishingly small chance we'll remember to update this comment, and then when we go to change these lines, we see this comment and wonder if the comment knows something we don't.

Reply:
Makes sense, I'll remove. Happy to avoid being on the wrong end of a `git blame` in the event something changes in the UI before we deprecate `/quartz/me`.